server/mediaserver.go

```diff
@@ -154,19 +162,59 @@ func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID str
 			mid = core.RandomManifestID()
 		}
 
+		var auth bool
+		if mid, auth = authenticateStream(url.String(), string(mid)); !auth {
```

Comment:
I think the flow clearer if we called `authenticateStream` first, before the manifestID checks. This also avoids having to pass in the manifest ID itself, which makes `authenticateStream` easier to understand (single responsibility). Also, returning an error rather than bool is more helpful.

```golang
mid, err := authenticateStream(url)
if err != nil {
    glog.Error("Could not authenticate stream: ", err)
    return ""
}
if mid == "" {
    mid = parseManifestID(url.Query().Get("manifestID"))
}
if mid == "" {
  mid = core.RandomManifestID()
}
```

Reply:
done